packages/components/src/components/files/WebDav.ts

```diff
@@ -0,0 +1,283 @@
+        return ActionURL.getContextPath() + '/_webdav' + ActionURL.encodePath(this.containerPath) + "/" + encodeURIComponent("@files");
+    }
+
+    // N.B.  We don't actually retrieve any of the custom properties here (since we won't display them), but need the
+    // rowId to be able to update descriptions reliably.
+    getCustomProperties(webDavFiles: Map<string, WebDavFile>) : Promise<Map<string, WebDavFile>> {
+        return new Promise((resolve, reject) => {
+            return Ajax.request({
+                url: ActionURL.buildURL('fileContent', 'getCustomProperties.api', this.containerPath),
+                method: "GET",
+                success: Utils.getCallbackWrapper((response) => {
+                    let updatedFiles = webDavFiles.asMutable();
+                    response.rows.forEach((row) => {
+                        if (updatedFiles.has(row['name'])) {
+                            updatedFiles.setIn([row['name'], 'propertiesRowId'], row['rowId']);
+                        }
+                    });
+                    resolve(updatedFiles.asImmutable());
+                }),
+                failure: Utils.getCallbackWrapper((response) => {
+                    console.error("Problem retrieving file content custom properties for container " + this.containerPath, response);
+                    reject(response)
+                }, null, false)
+            })
+        });
+    }
+
+    getWebDavFiles(directory?: string, includeDirectories?: boolean): Promise<Map<string, WebDavFile>> {
+        return new Promise((resolve, reject) => {
+            let url = this.getWebDavBaseUrl();
+
+            if (directory) {
+                url  += ("/" + directory);
+            }
+
+            return Ajax.request({
+                url: url + "?method=JSON",
+                method: "GET",
+                success: Utils.getCallbackWrapper((response) => {
+                    let filteredFiles = Map<string, WebDavFile>().asMutable();
+                    response.files.forEach((file) => {
+                        let webDavFile = WebDavFile.create(file);
+                        if (includeDirectories || !webDavFile.isCollection) {
+                            filteredFiles.set(webDavFile.name, webDavFile);
+                        }
+                    });
+                    resolve(filteredFiles.asImmutable());
+                }),
+                failure: Utils.getCallbackWrapper((response) => {
+                    console.error("Problem retrieving webDav files for container " + this.containerPath);
+                    reject(response)
+                }, null, false)
+            })
+        });
+    }
+
+    getFiles(includeDirectories?: boolean): Promise<Map<string, WebDavFile>> {
+        return new Promise((resolve, reject) => {
+            return this.getWebDavFiles(undefined, includeDirectories)
+                .then(response =>
+                    this.getCustomProperties(response)
```

Comment:
Why is the first `then` call split across two lines? it seems short enough to be on one line.